File: src/test/java/com/example/modulithgraalbug/ModularityTests.java
package com.example.modulithgraalbug;

import org.junit.jupiter.api.Test;
import org.springframework.modulith.core.ApplicationModules;

class ModularityTests {

    private ApplicationModules modules = ApplicationModules.of(Application.class);

    @Test
    void verifiesModularStructure() {
        modules.verify();
    }

}

File: src/test/kotlin/com/example/modulithgraalbug/ModularityTests.kt
package com.example.modulithgraalbug

import org.junit.jupiter.api.Test
import org.springframework.modulith.core.ApplicationModules

class ModularityTests {
    var modules: ApplicationModules = ApplicationModules.of(Application::class.java)

    @Test
    fun verifiesModularStructure() {
        modules.verify()
    }
}
